libyul/optimiser/Semantics.cpp

```diff
@@ -94,6 +94,19 @@ bool MSizeFinder::containsMSize(Dialect const& _dialect, Block const& _ast)
 	return finder.m_msizeFound;
 }
 
+bool MSizeFinder::containsMSize(Dialect const& _dialect, Object const& _object)
+{
+	if (containsMSize(_dialect, *_object.code))
```

Comment:
Should we add a recursion guard here?

Reply:
`RecursionGuard` is defined as protected in `ParserBase`. This class does not inherit from it.

Also, I don't think we really need it here. Objects cannot have loops.